### ISSUE TITLE
A bug in mutter is causing issues with window previews in some extensions

### ISSUE THREAD

**Issue by Krutonium** [NixOS/nixpkgs#169002]
### Describe the bug
I'm running my system on Unstable right now as one does on occasion, and I've run into a bug with an extension that I use called dash to panel. I tracked down the [bug on their tracker](https://github.com/home-sweet-gnome/dash-to-panel/issues/1620), and found that it's a bug in Mutter itself, that was fixed in [these two commits](https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2321/commits).

I am writing to request that these two commits be added to mutter, basically, and I'll admit I'm not even sure this is the place to do it, or the way to do it. If it's not, point me where to/how to, please and thank you!

### Steps To Reproduce
1. Use Gnome 42
2. Install Dash to Dock
3. Window Previews don't close due to a bug in Mutter

### Expected behavior
Window Previews should close

### Additional context
More information can be found at the issue I linked previously

### Notify maintainers

Basically the @gnome group which [appears to be made up of](https://github.com/NixOS/nixpkgs/blob/master/maintainers/team-list.nix#L245-L252)
      @bobby285271
      @hedning
      @jtojnar
      @dasj19
      @maxeaubrey

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.2-zen3, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.0pre20220411_f7276bc`
 - channels(krutonium): `""`
 - channels(root): `"nixos-21.11.335665.0f316e4d72d, nixos-hardware"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

**Comment by jtojnar** [comment]
If it bothers you enough, you can open a PR against mutter, adding the commits using `fetchpatch`:

https://github.com/NixOS/nixpkgs/blob/1ffba9f2f683063c2b14c9f4d12c55ad5f4ed887/pkgs/desktops/gnome/core/mutter/default.nix#L66 
https://github.com/NixOS/nixpkgs/blob/1ffba9f2f683063c2b14c9f4d12c55ad5f4ed887/pkgs/desktops/gnome/core/gnome-shell/default.nix#L96

Hopefully, the patch will apply without any changes.

Otherwise, you can wait for 42.1 release, for which tarballs should be available on Saturday, April 23:

https://wiki.gnome.org/FortyTwo

**Comment by Krutonium** [comment]
When is 42.1 coming to unstable out of curiosity?

**Comment by jtojnar** [comment]
Unfortunately, it still have not been released yet: https://gitlab.gnome.org/GNOME/mutter/-/tags

**Comment by jtojnar** [comment]
Opened https://github.com/NixOS/nixpkgs/pull/171904

**Comment by Krutonium** [comment]
This seems to be fixed! Thanks :)
